File: simple/python/parameters.py
import numpy as np
from randomwalkincome import randomwalkincome

# Model Parameters
N = 20		# number of grid points of assets
I = 10		# number of grid points of permanent income
T = 45		# number of periods total
R = 15		# number of periods in retirement

# Income
inc_work = 15000*np.ones((1, T-R))
inc_ret = 5000*np.ones((1,R))
inc = np.hstack((inc_work, inc_ret))

# Labor Supply
work = np.hstack((np.ones((1, T-R)), np.zeros((1, R))))

# Random Income Component
sigma = 0.5
TransP, RI = randomwalkincome(I, sigma)
RandInc = np.exp(RI)
# Income variables

inc_unc = np.ones((I, T))
for i in xrange(I):
	print
	for t in xrange(T):
		inc_unc[i,t] = max(0, inc[0,t]*RandInc[i])

# Other Parameters
beta = 0.98
r = 0.03
gamma = 1.5
psi = 0.003

# Wealth Grid
WealthGrid = np.linspace(-200000, 2000000, N).reshape((N, 1))

#Find where grid is 0
a0 = abs(WealthGrid).argmin()

File: simple/matlab/parameters.m
global gamma psi

%%%%SET UP THE PARAMETERS OF THE MODEL

N = 1000;                                    % # of grid points of assets
I=10;                                       % # of grid points of permanent income
T = 45;                                     % # of periods in totakl
R = 15;                                     % # of periods in retirement
B=2;                                        % possible assets liquidation

%%%%%%%%%%%% FIXED PARAMETERS %%%%%%%%

%%%%%%income
Income=15000*ones(1,T-R);
Income1=5000*ones(1,R);
Income=[Income,Income1];

%%%%%5 kabor supply
Works=[ones(1,T-R),zeros(1,R)];

%%%%%% RANDOM INCOME COMPONENT
sigma=0.5;
[TransP,RI]=randomwalkincome(I,sigma);

RandInc=exp(RI);

%%%%%%INCOME VARIABLES
for i=1:I,
    for t=1:T,
        Income_unc(i,t)=max(0,Income(t)*RandInc(i));
    end;
end;


%Other parameters
beta = single(0.98);                    % discount factor
r = single(0.03);                       % return on Assets
gamma = single(1.5);                    % RRA
psi = 0.003;                              % utility cost of working

%%%%%%%%% Wealth Grid 
WealthGrid = single(linspace(-2000000,2000000,N)');    % N equal spaced grid points between a and b'
%%%% find on what point on this grid the 0 is
[~,a0]=min(abs((WealthGrid)));

WealthGrid=WealthGrid(a0:end);
N=numel(WealthGrid);

%%%%%% asset value
AssetValue=zeros(T,1);
AV0=0;
AV1=1000;
AV2=-25;
for t=1:T,
    AssetValue(t)=max(AV0+AV1*t+AV2*t^2,0);
end;
